core/domain/exp_domain.py

```diff
@@ -2033,6 +2041,50 @@ def _convert_states_v48_dict_to_v49_dict(cls, states_dict):
 
         return states_dict
 
+    @classmethod
+    def _convert_states_v49_dict_to_v50_dict(cls, states_dict):
+        """Converts from version 49 to 50. Version 50 removes rules from
+        explorations that use one of the following rules:
+        [ContainsSomeOf, OmitsSomeOf, MatchesWithGeneralForm]. It also renames
+        `customOskLetters` cust arg to `allowedVariables`.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for state_dict in states_dict.values():
+            if state_dict['interaction']['id'] in MATH_INTERACTION_TYPES:
+                filtered_answer_groups = []
+                for answer_group_dict in state_dict[
+                        'interaction']['answer_groups']:
+                    filtered_rule_specs = []
+                    for rule_spec_dict in answer_group_dict['rule_specs']:
+                        rule_type = rule_spec_dict['rule_type']
+                        if rule_type not in MATH_INTERACTION_DEPRECATED_RULES:
+                            filtered_rule_specs.append(
+                                copy.deepcopy(rule_spec_dict))
+                    answer_group_dict['rule_specs'] = filtered_rule_specs
+                    if len(filtered_rule_specs) > 0:
+                        filtered_answer_groups.append(
+                            copy.deepcopy(answer_group_dict))
+                state_dict[
+                    'interaction']['answer_groups'] = filtered_answer_groups
```

Comment:
What happens if a published exploration becomes invalid due to this? Is there an audit job for this?

Reply:
We have already ran an audit job for such cases.